### PR TITLE
i915.c: Add NV12 and camera usage as supported combination

### PR DESCRIPTION
**Pull request by saranyagopal1**
Intel platforms use NV12 pixel format for camera.
So, add the combinations with NV12 format and usage flags
for camera in i915 driver.

Change-Id: Ifaec876139eff593af3360db79117160b36551b8
Signed-off-by: saranya <saranya.gopal@intel.com>

### CODE REVIEW

**Comment by saranyagopal1** [comment]
@kalyankondapally @JeevakaPrabu

**Comment by saranyagopal1** [comment]
Tested on v0.2 gralloc without enabling gralloc1.0.

File: i915.c [hunk 1]
```diff
@@ -119,6 +121,34 @@ static int i915_add_combinations(struct driver *drv)
...
+	ret = drv_add_combination(drv, DRM_FORMAT_NV12, &metadata, read_camera_flags);
+	if (ret)
+		return ret;
+
+	ret = drv_add_combination(drv, DRM_FORMAT_NV12, &metadata, write_camera_flags);
+	if (ret)
+		return ret;
+
+	ret = drv_add_combination(drv, DRM_FORMAT_NV12, &metadata,
+				BO_USE_HW_CAMERA_ZSL | read_camera_flags);
```
Comment (kalyankondapally) [inline i915.c]: This can go to read_camera_flags and write_camera_flags, no need to have another couple of calls here?

File: i915.c [hunk 2]
```diff
@@ -119,6 +121,34 @@ static int i915_add_combinations(struct driver *drv)
...
+	if (ret)
+		return ret;
+
+	ret = drv_add_combination(drv, DRM_FORMAT_NV12, &metadata,
+				BO_USE_HW_CAMERA_ZSL | write_camera_flags);
+	if (ret)
+		return ret;
+
+	ret = drv_add_combination(drv, DRM_FORMAT_NV12, &metadata,
+				BO_USE_TEXTURE | read_camera_flags);
```
Comment (kalyankondapally) [inline i915.c]: this can be part of read_camera_flags above

File: i915.c [hunk 3]
```diff
@@ -119,6 +121,34 @@ static int i915_add_combinations(struct driver *drv)
...
+	if (ret)
+		return ret;
+
+	ret = drv_add_combination(drv, DRM_FORMAT_NV12, &metadata,
+			BO_USE_TEXTURE | BO_USE_HW_CAMERA_ZSL | read_camera_flags);
```
Comment (kalyankondapally) [inline i915.c]: u will not need this if above changes are done

**Comment by saranyagopal1** [comment]
Please hang on. I am modifying this again to use only BO_CAMERA_USAGE_MASK which I am going to define it in drv_priv.h.

**Comment by saranyagopal1** [comment]
Changing the implementation and submitting a new one.

**Pull request closed without merging**